core/src/main/java/io/questdb/griffin/engine/functions/metadata/BuildFuntionFactory.java

```diff
@@ -0,0 +1,93 @@
+/*******************************************************************************
+ *     ___                  _   ____  ____
+ *    / _ \ _   _  ___  ___| |_|  _ \| __ )
+ *   | | | | | | |/ _ \/ __| __| | | |  _ \
+ *   | |_| | |_| |  __/\__ \ |_| |_| | |_) |
+ *    \__\_\\__,_|\___||___/\__|____/|____/
+ *
+ *  Copyright (c) 2014-2019 Appsicle
+ *  Copyright (c) 2019-2020 QuestDB
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ ******************************************************************************/
+
+package io.questdb.griffin.engine.functions.metadata;
+
+import io.questdb.BuildInformation;
+import io.questdb.cairo.CairoConfiguration;
+import io.questdb.cairo.sql.Function;
+import io.questdb.griffin.FunctionFactory;
+import io.questdb.griffin.SqlException;
+import io.questdb.griffin.SqlExecutionContext;
+import io.questdb.griffin.engine.functions.StrFunction;
+import io.questdb.griffin.engine.functions.constants.StrConstant;
+import io.questdb.std.Misc;
+import io.questdb.std.ObjList;
+import io.questdb.std.str.CharSink;
+import io.questdb.std.str.StringSink;
+
+public class BuildFuntionFactory implements FunctionFactory {
+    @Override
+    public String getSignature() {
+        return "build()";
+    }
+
+    @Override
+    public Function newInstance(final ObjList<Function> args,
+                                final int position,
+                                final CairoConfiguration configuration,
+                                final SqlExecutionContext sqlExecutionContext) throws SqlException {
+
+        return Memoizer.getInstance(configuration).function;
+    }
+
+    private static class Memoizer {
+        private static Memoizer instance;
+
+        private final StrFunction function;
+
+        private Memoizer(final CharSequence message) {
+            this.function = new StrConstant(0, message);
+        }
+
+        public static Memoizer getInstance(final CairoConfiguration configuration) {
+            if (instance == null) {
```

Comment:
this is some heavy artillery going on here :)

Generally speaking FunctionFactory instances are singletons already, this could perhaps remove static variable use. Then double-check locking can perhaps be replaced with 

```java
if (instance == null) {
    instance = new Memoizer(message(configuration));
}
```

The worst case scenario is that under very heavy load we might have several classes created unnecessarily. But realistically this is very small and unlikely overhead in the name of code simplicity.

Reply:
Then we don't even need the Memoizer. The instance will be a StrFunction.

In regards to thread safety, if I'm understanding correctly, you are saying to do the same that is done when computing the hash of a string, right?